lambda/lambda.mli

```diff
@@ -38,15 +38,23 @@ type alloc_mode = private
   | Alloc_heap
   | Alloc_local
 
+type modify_mode = private
```

Comment:
I'm unsure why the change to add this `modify_mode` is in this PR.  Could we split this into a separate PR?